packages/react-primitives/src/components/read-only/index.tsx

```diff
@@ -0,0 +1,79 @@
+import { Copy } from "../icon/copy";
+import clsx from "clsx";
+import * as styles from "./read-only.css";
+import { useEffect, useRef } from "react";
+
+type ReadOnlyProps = {
+  value: string;
+  id: string;
+  label?: string;
+  as?: "input" | "textarea";
+  rows?: number;
+  copy?: boolean;
+  className?: string;
+};
+
+export function ReadOnly({
```

Comment:
Can you please specify what is the use case of this component? Some JS-doc style explanations here would be helpful, it's not easy to understand from the code